test/e2e/workflow_test.go

```diff
@@ -274,5 +272,93 @@ var _ = Describe("Workflow Persistence Use Cases :: ", Label("flows-persistence"
 		Entry("defined from the sonataflow platform as reference and without DI and JS", test.GetPathFromE2EDirectory("workflows", "persistence", "from_platform_without_di_and_js_services"), true, false),
 		Entry("defined from the sonataflow platform as reference but not required by the workflow", test.GetPathFromE2EDirectory("workflows", "persistence", "from_platform_with_no_persistence_required"), false, false),
 	)
+})
+
+var _ = Describe("Workflow Monitoring Use Cases :: ", Label("flows-monitoring"), Ordered, func() {
+
+	var targetNamespace string
+	BeforeEach(func() {
+		targetNamespace = fmt.Sprintf("test-%d", rand.Intn(randomIntRange)+1)
+		err := kubectlCreateNamespace(targetNamespace)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	AfterEach(func() {
+		// Remove resources in test namespace
+		if !CurrentSpecReport().Failed() && len(targetNamespace) > 0 {
+			cmd := exec.Command("kubectl", "delete", "sonataflow", "--all", "-n", targetNamespace, "--wait")
+			_, err := utils.Run(cmd)
+			Expect(err).NotTo(HaveOccurred())
+			err = kubectlDeleteNamespace(targetNamespace)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+	Describe("basic workflow monitoring", func() {
+		projectDir, _ := utils.GetProjectDir()
 
+		It("should create servicemonitor for workflow deployed as k8s deployment when monitoring enabled in platform", func() {
+			By("Deploy the SonataFlowPlatform CR and the workflow")
+			var manifests []byte
+			EventuallyWithOffset(1, func() error {
+				var err error
+				cmd := exec.Command("kubectl", "kustomize", filepath.Join(projectDir,
+					"test/e2e/testdata/workflows/prometheus", "k8s_deployment"))
+				manifests, err = utils.Run(cmd)
+				return err
+			}, time.Minute, time.Second).Should(Succeed())
+			cmd := exec.Command("kubectl", "create", "-n", targetNamespace, "-f", "-")
```

Comment:
After creating the workflow, we then replace the image in the CR. Please see the other examples.